libmscore/measure.cpp

```diff
@@ -474,57 +474,70 @@ qreal Measure::tick2pos(Fraction tck) const
                               (score()->styleI(Sid::measureNumberInterval) == 1);
                         }
                   }
             }
-      QString s;
-      if (smn)
-            s = QString("%1").arg(no() + 1);
+      QString measureNumber;
+      if (show)
+            measureNumber.setNum(no() + 1);
       int nn = 1;
-      bool nas = score()->styleB(Sid::measureNumberAllStaffs);
-
-      if (!nas) {
-            //find first non invisible staff
-            for (unsigned staffIdx = 0; staffIdx < _mstaves.size(); ++staffIdx) {
-                  MStaff* ms = _mstaves[staffIdx];
-                  SysStaff* ss  = system()->staff(staffIdx);
-                  Staff* staff = score()->staff(staffIdx);
-                  if (ms->visible() && staff->show() && ss->show()) {
-                        nn = staffIdx;
-                        break;
+      bool numberAllStaves = score()->styleB(Sid::measureNumberAllStaves);
+
+      if (!numberAllStaves) { // number one staff by system
+            if (score()->styleV(Sid::measureNumberSystemPlacement) == int(Placement::ABOVE)) {
+                  //find first non invisible staff
+                  for (unsigned staffIdx = 0; staffIdx < _mstaves.size(); ++staffIdx) {
+                        MStaff* ms = _mstaves[staffIdx];
+                        SysStaff* ss  = system()->staff(staffIdx);
+                        Staff* staff = score()->staff(staffIdx);
+                        if (ms->visible() && staff->show() && ss->show()) {
+                              nn = staffIdx;
+                              break;
+                              }
+                        }
+                  }
+            else {
+                  //find last non invisible staff
+                  for (int staffIdx = _mstaves.size() - 1; staffIdx > -1; --staffIdx) {
+                        MStaff* ms = _mstaves[staffIdx];
+                        SysStaff* ss  = system()->staff(staffIdx);
+                        Staff* staff = score()->staff(staffIdx);
+                        if (ms->visible() && staff->show() && ss->show()) {
+                              nn = staffIdx;
+                              break;
+                              }
                         }
                   }
             }
       for (int staffIdx = 0; staffIdx < int(_mstaves.size()); ++staffIdx) {
             MStaff* ms       = _mstaves[staffIdx];
             MeasureNumber* t = ms->noText();
             if (t)
-                  t->setTrack(staffIdx * VOICES);
-            if (smn && ((staffIdx == nn) || nas)) {
+                  t->setTrack(staffIdx* VOICES);
+            if (show && ((staffIdx == nn) || numberAllStaves)) {
                   if (t == 0) {
                         t = new MeasureNumber(score());
-                        t->setTrack(staffIdx * VOICES);
+                        t->setTrack(staffIdx* VOICES);
```

Comment:
Please don't do this. It's part of the coding principles to put spaces both in front of and after an operator.

Reply:
wow, really thought it was a pointer...